liberty-maven-plugin/src/it/generate-features-it/src/test/java/net/wasdev/wlp/test/dev/it/GenerateFeaturesTest.java

```diff
@@ -55,4 +56,28 @@ public void basicTest() throws Exception {
         assertEquals(expectedFeatures, features);
     }
 
+    @Test
+    public void customFeaturesTest() throws Exception {
+        // complete the setup of the test
+        File serverXmlFile = new File(tempProj, "src/main/liberty/config/server.xml");
+        replaceString("<!--replaceable-->",
+        "<featureManager>\n" +
+        "  <feature>servlet-4.0</feature>\n" +
+        "  <feature>usr:custom-1.0</feature>\n" +
+        "</featureManager>\n", serverXmlFile);
+        File newFeatureFile = new File(tempProj, GENERATED_FEATURES_FILE_PATH);
+        assertFalse("Before running", newFeatureFile.exists());
+        // run the test
+        runProcess("mvn compile liberty:generate-features");
+
+        // verify that the generated features file was created
+        assertTrue(getLogTail(), newFeatureFile.exists());
+
+        // verify that the correct feature is in the generated-features.xml
+        List<String> features = readFeatures(newFeatureFile);
+        assertEquals(getLogTail(), 1, features.size());
```

Comment:
Once you pull latest from main, rather than `getLogTail()` you can just print `processOutput`, see:

https://github.com/OpenLiberty/ci.maven/blob/f64fa90f7626c5f2f8c6130fe51753115e49cfd6/liberty-maven-plugin/src/it/generate-features-it/src/test/java/net/wasdev/wlp/test/dev/it/GenerateFeaturesTest.java#L73

https://github.com/OpenLiberty/ci.maven/blob/f64fa90f7626c5f2f8c6130fe51753115e49cfd6/liberty-maven-plugin/src/it/generate-features-it/src/test/java/net/wasdev/wlp/test/dev/it/BaseGenerateFeaturesTest.java#L117-L119

Reply:
Lucky there were conflicts requiring me to rewrite. Thank you.